src/builder-utils.c

```diff
@@ -883,7 +883,7 @@ builder_host_spawnv (GFile                *dir,
   g_source_remove (sigint_id);
   g_dbus_connection_signal_unsubscribe (connection, subscription);
 
-  if (!g_spawn_check_exit_status (data.exit_status, error))
+  if (!g_spawn_check_wait_status (data.exit_status, error))
```

Comment:
This is a much newer API, we probably don't want to require this version of GLIB.

Instead what we should do is define `-DGLIB_VERSION_MIN_REQUIRED=GLIB_VERSION_2_44 -DGLIB_VERSION_MAX_ALLOWED=GLIB_VERSION_2_44` in our build system. This will silence the deprecation warnings.

Reply:
```
AC_DEFINE([GLIB_VERSION_MIN_REQUIRED], [GLIB_VERSION_2_66], [Glib min version])
AC_DEFINE([GLIB_VERSION_MAX_ALLOWED], [GLIB_VERSION_2_66], [Glib max version])
```

Defining those two macros result in an 'Not available before 2.68' warning for `g_memdup2`. Maybe something we can fix in libglnx?

But it also throws the same warning for `G_URI_FLAGS_SCHEME_NORMALIZE`, that got just introduced with the libsoup changes, but the min version got only bumped to version 2.66. I couldn't find any signs of a backport therefore I think we should increase the version to 2.68. 

If that's the case should I remove the commit with the libglnx bump?

Reply:
> But it also throws the same warning for G_URI_FLAGS_SCHEME_NORMALIZE, that got just introduced with the libsoup changes, but the min version got only bumped to version 2.66.

It is in a `#if GLIB_CHECK_VERSION (2, 68, 0)` check block so its fine.

Let's just do `GLIB_VERSION_MAX_ALLOWED=GLIB_VERSION_2_68` and leave the MIN.

Reply:
Now I see, thanks.

Leaving the min version completely out didn't work, so I set it to 2.66.

But maybe it is better to just drop the changes, the commit complicates the build config only to silence a compiler warning. If setting those macros doesn't have any other benefits I'm fine with dropping this commit. What do you think?

Reply:
The macros are great! It will prevent future slip ups when using too new of APIs.